### PR TITLE
Refactor for DI FusionJS

### LINKED ISSUE #21: Refactor for DI FusionJS

### PR DESCRIPTION
**Pull request by angus-c**
Resolves #21

### CODE REVIEW

File: .cuprc.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
```
Comment (KevinGrandon) [inline .cuprc.js]: Is this needed? Do we actually use React here?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,72 @@
+{
+  "name": "fusion-plugin-service-worker",
+  "description": "A Fusion.js plugin",
```
Comment (KevinGrandon) [inline package.json]: nit: update description.

File: src/handlers.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+function responseIsHtml(response) {
+  if (!response || !response.headers) {
+    return false;
+  }
+  const contentType = response.headers.get('content-type');
+  return contentType && contentType.indexOf('html') > -1;
+}
+
+/*
+// usage in src/sw-handler.js
```
Comment (KevinGrandon) [inline src/handlers.js]: Seems like we should remove this and instead document this inside the README?

File: src/types.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  from: (
+    ctx?: Context
+  ) => {
+    ctx?: Context,
+    value: string,
+  },
+};
+
+export type AssetInfo = {
+  // TODO: want these to be Array<string> but `caches.addAll` expects array<T>
```
Comment (KevinGrandon) [inline src/types.js]: It seems like this is ` Array<string>` now? Is this TODO accurate?

File: .cuprc.js (hunk 1 above)
Comment (angus-c) [reply .cuprc.js]: It's not—good catch

File: src/handlers.js (hunk 3 above)
Comment (angus-c) [reply src/handlers.js]: Sure

**Comment by KevinGrandon** [comment]
Something weird happened with the history here, not sure why it's showing all of the previous commits on master? I bet if we rebase against upstream/master, you could probably drop all of the previous commits?

File: src/handlers.js [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+                )
+            );
+          })
+          .catch(e => {
+            throw new Error(`sw: error updating cache ${cacheName}: ${e}`);
+          })
+      );
+    },
+    onFetch: (event: FetchEvent) => {
+      const HTML_TTL = 1 * 24 * 60 * 60 * 1001; // 1 day
```
Comment (rajeshsegu) [inline src/handlers.js]: move this to the top.

File: src/mock-service-worker-source.js [hunk 6]
```diff
@@ -0,0 +1,113 @@
+export default `var serviceWorker = function({precachePaths, cacheablePaths}) {
```
Comment (rajeshsegu) [inline src/mock-service-worker-source.js]: rename file to service-worker-template.js

**Comment by angus-c** [comment]
Closing, too far gone to rebase against master

**Pull request closed without merging**